sogs/model.py

```diff
@@ -541,14 +1065,33 @@ def set_moderator(self, *, admin=False, visible=False):
         self.global_moderator = True
         self.visible_mod = visible
 
-    def remove_moderator(self):
+    def remove_moderator(self, *, removed_by: User):
         """Removes this user's global moderator/admin status, if set."""
+
+        if not removed_by.global_admin:
```

Comment:
what if removed_by is None?

Reply:
Then you called it wrong and deserve the exception that you get.